### ISSUE TITLE
[Bug]: PR-curve generated with AUPR metric uses wrong axis (and title?)

### ISSUE THREAD

**Issue by Cral0202** [open-edge-platform/anomalib#2636]
### Describe the bug

When you generate a PR-curve with:
```
aupr = AUPR(fields = ["pred_score", "gt_label"])
aupr(batch)
fig, title = aupr.generate_figure()
```
the recall appears on the Y-axis, and the precision appears on the X-axis. To my knowledge, this is not the standard way to do it. It also seems incorrect, because the lines in the graph assume that recall is on the X-axis and precision is on the Y-axis.

You can see this from the 2 figures. One is generated with ```fig, title = aupr.generate_figure()```, and the other is generated with PLT and BinaryPrecisionRecallCurve. This dataset consists of 58% positive samples. Looking at the "PR" figure, at a 100% recall where all of the samples are predicted as positive, the precision is 58%. This makes sense. However, looking at the "AUPR" figure, a 100% recall yields a 0% precision. I don't see how that makes sense.

Furthermore, when you generate a ROC curve with the AUROC metric, the title of the curve becomes "ROC". But when you generate a PR curve with the AUPR metric, the title of the curve becomes "AUPR". This doesn't seem consistent. Shouldn't the title become "PR"?

![Image](https://github.com/user-attachments/assets/12e68d76-00cc-4a2a-ae4e-5da81386f593)

![Image](https://github.com/user-attachments/assets/2e36f9cc-e56d-4bb6-b914-372b62e0b0ef)

### Dataset

Folder

### Model

PatchCore

### Steps to reproduce the behavior

The code used, some is irrelevant. Using classification. Predict with ```predictions = engine.predict()```, then:

```
if predictions is not None:
    predictions_list = []
    predictions_list_binary = []
    labels_list = []
    images_list = []
    num_of_samples = len(predictions)
    correct_predictions = 0
    incorrect_predictions = 0
    norm_as_anom = 0
    anom_as_norm = 0

    for prediction in predictions:
        image_path = prediction.image_path
        pred_label = prediction.pred_label
        pred_score = prediction.pred_score
        image_tensor = prediction.image

        folder_name = os.path.basename(os.path.dirname(image_path[0]))

        is_normal = folder_name.lower() == "ok"
        true_label = not is_normal  # If normal, label should be False (0), otherwise True (1)

        # Append the prediction score and ground truth label
        predictions_list.append(pred_score)
        predictions_list_binary.append(pred_label)
        labels_list.append(true_label)
        images_list.append(image_tensor)

        # Check which ones were incorrectly classified
        if is_normal == pred_label:
            incorrect_predictions += 1
            
            if is_normal:
                msg = "Normal sample incorrectly predicted as anomalous:"
                norm_as_anom += 1
            else:
                msg = "Anomalous sample incorrectly predicted as normal:"
                anom_as_norm += 1
                
            print(msg, image_path, pred_label, pred_score, sep="\n", end="\n\n")
        else:
            correct_predictions += 1

    # Convert lists to tensors for calculation
    preds = torch.tensor(predictions_list)
    target = torch.tensor(labels_list)
    images_tensor = torch.stack(images_list).squeeze(1)  # Remove the extra dimension

    # Print results
    print(f"Number of samples: {num_of_samples}")
    print(f"Number of correct predictions: {correct_predictions}")
    print(f"Number of incorrect predictions: {incorrect_predictions} \n")
    print(f"Number of normal samples incorrectly predicted as anomalous: {norm_as_anom}")
    print(f"Number of anomalous samples incorrectly predicted as normal: {anom_as_norm}\n")
```

```
batch = ImageBatch(
    image = images_tensor,
    pred_score = preds,
    gt_label = target
)
```

```
aupr = AUPR(fields = ["pred_score", "gt_label"])
aupr(batch)
fig, title = aupr.generate_figure()
```

### OS information

OS information:
- OS: Windows 11
- Python version: 3.12.3
- Anomalib version: 2.0
- PyTorch version: 2.6.0
- CUDA/cuDNN version: 12.4
- GPU models and configuration: 1x GeForce RTX 3060 Ti
- Any other relevant information: I use Jupyter Notebook

### Expected behavior

Recall would be shown on the X-axis, and precision would be shown on the Y-axis. The title would also show "PR" or similar instead of "AUPR".

### Pip/GitHub

pip

### Configuration YAML

```yaml
N/A
```

### Logs

```shell
N/A
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by Cral0202** [comment]
Could probably just go into "aupr.py" and change:

```
xlabel = "Precision"
ylabel = "Recall"
title = "AUPR"
```

to:

```
xlabel = "Recall"
ylabel = "Precision"
title = "PR"
```
